### PR TITLE
 [AIRFLOW-5830] Get rid of slim image.

### PR DESCRIPTION
**Pull request by potiuk**
The slim image gave only very small gain on executing the tests in CI. The
image was significantly smaller, but then for local development and testing
you needed both full CI and SLIM-CI image.

This made the scripts and docker image needlessly complex - especially
in the wake of coming Production image it turned to be premature
optimisation really. While it sped-up (slightly - by 10-20 seconds) some
static check jobs in Travis, it increased time needed by developers
to have a working environment and to keep it updated every time it was
needed (by minutes)

Also having two separately managed images made it rather complex to join
some of the Travis CI jobs (there is a follow-up change with getting rid
of Checklicence image).

With this change both static checks and tests are executed using single
image. That also opens doors for further simplification of the scripts
and easier implementation of production image.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5826

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: BREEZE.rst [hunk 1]
```diff
@@ -79,11 +79,9 @@ images maintained on the Docker Hub in the ``apache/airflow`` repository.
 
 There are three images that we are currently managing:
 
-* **Slim CI** image that is used for static code checks (size of ~500MB). Its tag follows the pattern
-  of ``<BRANCH>-python<PYTHON_VERSION>-ci-slim`` (for example, ``apache/airflow:master-python3.6-ci-slim``).
-  The image is built using the `<Dockerfile>`_ Dockerfile.
-* **Full CI image*** that is used for testing. It contains a lot more test-related installed software
-  (size of ~1GB). Its tag follows the pattern of ``<BRANCH>-python<PYTHON_VERSION>-ci``
+* **Full CI image*** that is used for testing od both Unit tests and static check tests.
```
Comment (mik-laj) [inline BREEZE.rst]: Maybe we should call this image "CI Image" (without full) now ?

File: scripts/ci/_utils.sh [hunk 2]
```diff
@@ -239,8 +239,6 @@ function update_all_md5_files() {
 # files. There are the following, problems with the current Dockerfiles that need longer build times:
 # 1) We need to fix group permissions of files in Docker because different linux build services have
```
Comment (mik-laj) [inline scripts/ci/_utils.sh]: This numbered list does not make sense now. The previous paragraph talks about many problems, and only one problem is described.

File: Dockerfile [hunk 3]
```diff
@@ -121,114 +113,95 @@ RUN adduser airflow \
...
-ARG KUBERNETES_VERSION="v1.15.0"
-ENV KUBERNETES_VERSION=${KUBERNETES_VERSION}
-ARG KIND_VERSION="v0.5.0"
-ENV KIND_VERSION=${KIND_VERSION}
+RUN mkdir -pv "${HADOOP_HOME}" \
+    && mkdir -pv "${HIVE_HOME}" \
+    && mkdir -pv "${MINICLUSTER_HOME}" \
+    && mkdir -pv "/user/hive/warehouse" \
+    && chmod -R 777 "${HIVE_HOME}" \
+    &&chmod -R 777 "/user/"
```
Comment (mik-laj) [inline Dockerfile]: Missing space.

File: Dockerfile [hunk 4]
```diff
@@ -121,114 +113,95 @@ RUN adduser airflow \
...
+  && chmod +x kubectl \
+  && mv kubectl /usr/local/bin/kubectl
 
-SHELL ["/bin/bash", "-o", "pipefail", "-e", "-u", "-x", "-c"]
+RUN curl -Lo kind \
+   "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-linux-amd64" \
+   && chmod +x kind \
+   && mv kind /usr/local/bin/kind
 
 RUN echo "Airflow version: ${AIRFLOW_VERSION}"
```
Comment (mik-laj) [inline Dockerfile]: This line is duplicated on 31

File: hooks/build [hunk 5]
```diff
@@ -412,21 +373,18 @@ elif [[ "${AIRFLOW_CONTAINER_USE_PULLED_IMAGES_CACHE}" == "true" ]]; then
...
     echo
     echo "Use default cache from locally built images."
     echo
     echo "Cache directives used: "
     set +u
     echo "CI build: ${DOCKER_CACHE_DIRECTIVE_CI[*]}"
-    echo "CI slim build: ${DOCKER_CACHE_DIRECTIVE_CI_SLIM[*]}"
     echo "CI checklicence build: ${DOCKER_CACHE_DIRECTIVE_CHECKLICENCE[*]}"
     set -u
     echo
```
Comment (mik-laj) [inline hooks/build]: Should we add quote on 404 line?

Comment (mik-laj) [inline hooks/build:390]: Should we add quote on 404 line (new file)?

File: scripts/ci/_utils.sh [hunk 6]
```diff
@@ -575,10 +534,9 @@ function rebuild_checklicence_image_if_needed() {
 }
 
 #
-# Cleans up the CI slim image
+# Cleans up the CI image
```
Comment (mik-laj) [inline scripts/ci/_utils.sh]:
```diff
-# Cleans up the CI image
+# Cleans up the Checklicense image
```

**Review by mik-laj — APPROVED** [review]
It looks good, but I think another person should look to be sure.

**Review by Fokko — APPROVED** [review]
Love it @potiuk.

File: BREEZE.rst (hunk 1 above)
Comment (potiuk) [reply BREEZE.rst]: Yep.

File: scripts/ci/_utils.sh (hunk 2 above)
Comment (potiuk) [reply scripts/ci/_utils.sh]: True!

File: Dockerfile (hunk 3 above)
Comment (potiuk) [reply Dockerfile]: Yep

File: Dockerfile (hunk 4 above)
Comment (potiuk) [reply Dockerfile]: True

File: hooks/build (hunk 5 above)
Comment (potiuk) [reply hooks/build:390]: Why not

File: scripts/ci/_utils.sh (hunk 6 above)
Comment (potiuk) [reply scripts/ci/_utils.sh]: Will be gone in next one, but yes.

**Pull request merged**